h2/src/java9/src/org/h2/util/Bits.java

```diff
@@ -29,6 +29,22 @@
      */
     private static final VarHandle LONG_VH = MethodHandles.byteArrayViewVarHandle(long[].class, ByteOrder.BIG_ENDIAN);
 
+    /**
+     * Compare the contents of two char arrays. If the content or length of the
+     * first array is smaller than the second array, -1 is returned. If the content
+     * or length of the second array is smaller than the first array, 1 is returned.
+     * If the contents and lengths are the same, 0 is returned.
+     *
+     * @param data1
+     *            the first char array (must not be null)
+     * @param data2
+     *            the second char array (must not be null)
+     * @return the result of the comparison (-1, 1 or 0)
+     */
+    public static int compareNotNull(char[] data1, char[] data2) {
+        return Integer.signum(Arrays.compare(data1, data2));
```

Comment:
signum not neessary here?

Reply:
Only 1, -1 and 0 are allowed in H2's functions as a result of comparison for some reason. I remember that there were some failures when I wrote optimized Java 9 version for byte[] comparison without `Integer.signum()` some time ago. That's why I put `Integer.signum()` here too.

`Arrays.compare()` methods may return any integer value.